apps/prairielearn/src/lib/groups.sql

```diff
@@ -208,50 +201,50 @@ VALUES
 -- BLOCK reassign_group_roles_after_leave
 WITH
   deleted_group_users_roles AS (
-    DELETE FROM group_user_roles
+    DELETE FROM group_user_roles AS gur USING group_users AS gu
     WHERE
-      group_id = $group_id
+      gu.group_id = $group_id
+      AND gur.group_user_id = gu.id
   ),
   json_roles AS (
     SELECT
-      (role_assignment ->> 'user_id')::bigint AS user_id,
+      gu.id AS group_user_id,
       (role_assignment ->> 'group_role_id')::bigint AS group_role_id
     FROM
       JSON_ARRAY_ELEMENTS($role_assignments::json) AS role_assignment
+      JOIN group_users AS gu ON gu.group_id = $group_id
+      AND gu.user_id = (role_assignment ->> 'user_id')::bigint
   )
 INSERT INTO
-  group_user_roles (group_id, user_id, group_role_id)
+  group_user_roles (group_user_id, group_role_id)
```

Comment:
PENDING (and in other inserts): keep updating group_id AND user_id until this PR is deployed in all servers.

Reply:
Done in d9a0dce.